repository/lib/rom/repository/class_interface.rb

```diff
@@ -113,7 +113,13 @@ def define_restricted_command_method(type, views, **opts)
             if changeset.respond_to?(:commit)
               changeset.commit
             else
-              root.command(type, **opts).public_send(view_name, *view_args).call(*input)
+              #   in case of view_args been coercible to an array we do not want to splat it
+              #   example: Date and Time objects
+              if view_args.respond_to?(:to_a)
+                root.command(type, **opts).public_send(view_name, view_args).call(*input)
```

Comment:
What is the API for calling the command if the view has arity > 1? (There aren't any specs for such calls as far as I can see.)

Should the view args be wrapped in an array like so?:

```ruby
repo.delete_by_arity2_restriction([view_arg1, view_arg2], input)
```

(If it's not wrapped, I guess it would be splatted into `input` rather than the view_arg at the beginning of the defined command method?)

If it should be wrapped, could you maybe just wrap any non-array object in an array before splatting? Checking either by `#is_a?(Array)` or `#responds_to?(:to_ary)`:

```ruby
view_args = [view_args] unless view_args.responds_to?(:to_ary)
root.command(type, **opts).public_send(view_name, *view_args).call(*input)
```

Or even just by doing a splat reassignment:

```ruby
*view_args = view_args
root.command(type, **opts).public_send(view_name, *view_args).call(*input)
```

(Splat assignment only tries implicit conversion (`#to_ary`) and doesn't call `#to_a`, in contrast to the splatting of the argument on the next line which calls `#to_a` on view_args if `#to_ary` is not defined. But maybe this is too obscure to make the reason behind the code above clear.)